### PR TITLE
RFC: Document StreamArchive file format

### PR DESCRIPTION
**Pull request by aeikum**
I know of a handful of implementations of this format, so we should formally define the format. A couple notes:

-The version was stored big-endian. Since the MS three bytes are currently zero, I've re-defined it to be one byte, with the upper three bytes unused. We could instead make it four big-endian bytes, if you like.

-I've defined the tag and hash sizes to be flexible. Together they must occupy 40 bytes, but the hash can be an application-defined length, with the tag taking up the remainder.

### CODE REVIEW

File: fossilize_db.cpp [hunk 1]
```diff
@@ -610,27 +610,53 @@ DatabaseInterface *create_zip_archive_database(const char *path, DatabaseMode mo
...
+ * unused3         uint8_t        Currently unused. Must be zero.
+ * version         uint8_t        StreamArchive version: 6
+ *
+ *
+ * Each entry follows this format:
+ *
+ * Field           Type                              Description
+ * -----           ----                              -----------
+ * tag             unsigned char[40 - hash_bytes]    Application-defined 'tag' which groups entry types. Stored as hexadecimal big-endian ASCII.
+ * hash            unsigned char[hash_bytes]         Application-defined 'hash' to identify this entry. Stored as hexadecimal big-endian ASCII.
```
Comment (HansKristian-Work) [inline fossilize_db.cpp:635]: Endianness of the hash is technically application defined. It should be treated as 40 opaque hex values.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Sorry, didn't see this until now. Just a tiny nit, otherwise fine.

**Comment by HansKristian-Work** [comment]
Another thing to mention is that the stream can be cut off anywhere and the last invalid entry should be considered to not exist.

**Comment by aeikum** [comment]
Cool, good suggestions. I've pushed an update to fix those, and also mentions that tag+hash combos should be unique.

**Review by HansKristian-Work — APPROVED** [review]

**Pull request merged**